### PR TITLE
fix(w3c/headers): allow overriding latestVersion:

### LINKED ISSUE #3618: Allow latestVersion to be overridden by the author
**Is your feature request related to a problem? Please describe.**
For latestVersion, sometimes it's good to point directly to another URL. (e.g., for level specs ("/TR/foo-3/), sometimes you want latestVersion to be "TR/foo/" )

**Describe the solution you'd like**
latestVersion to take a URL... like

https://w3c.github.io/pointerevents/?latestVersion="pointerevents" (resolved against w3c/TR/!) 

**Describe alternatives you've considered**
Using `level` doesn't help. 

cc @plehegar @caribouW3

### PR DESCRIPTION
**Pull request by marcoscaceres**
Closes #3618

For latestVersion, sometimes it's good to point directly to another URL. (e.g., for level specs ("/TR/foo-3/), sometimes you want latestVersion to be "TR/foo/" )

### CODE REVIEW

File: src/w3c/headers.js [hunk 1]
```diff
@@ -63,7 +63,7 @@
 //      PAG on the document.
 //  - thisVersion: the URI to the dated current version of the specification. ONLY ever use this for CG/BG
 //      documents, for all others it is autogenerated.
-//  - latestVersion: the URI to the latest (undated) version of the specification. ONLY ever use this for CG/BG
+//  - latestVersion: the URI to the latest version of the specification.
 //      documents, for all others it is autogenerated.
```
Comment (saschanaz) [inline src/w3c/headers.js:67]: Should the next line be fixed too?

File: src/w3c/headers.js [hunk 2]
```diff
@@ -108,6 +108,17 @@ const W3CDate = new Intl.DateTimeFormat(["en-AU"], {
   day: "2-digit",
 });
 
+//
+/**
+ * Tagged template TR_URL`path`. Resolves against W3C's /TR/
+ * */
+function W3URL(strings, ...values) {
```
Comment (saschanaz) [inline src/w3c/headers.js:115]: I think typically template string functions have camel cases 👀

File: src/w3c/headers.js [hunk 3]
```diff
@@ -108,6 +108,17 @@ const W3CDate = new Intl.DateTimeFormat(["en-AU"], {
   day: "2-digit",
 });
 
+//
+/**
+ * Tagged template TR_URL`path`. Resolves against W3C's /TR/
```
Comment (saschanaz) [inline src/w3c/headers.js:113]: (Name doesn't match)

File: src/w3c/headers.js [hunk 4]
```diff
@@ -401,21 +417,20 @@ export function run(conf) {
...
-    const pmat = status2maturity[conf.previousMaturity]
-      ? status2maturity[conf.previousMaturity]
-      : conf.previousMaturity;
-    if (conf.isTagFinding) {
-      conf.prevVersion = `${conf.latestVersion}-${ISODate.format(
-        conf.previousPublishDate
-      )}`;
-    } else if (conf.isCGBG) {
+    const prevMaturity =
+      status2maturity[conf.previousMaturity] || conf.previousMaturity;
```
Comment (saschanaz) [inline src/w3c/headers.js:421]: Can we use `??` operator here?

File: src/w3c/headers.js [hunk 5]
```diff
@@ -108,6 +108,17 @@ const W3CDate = new Intl.DateTimeFormat(["en-AU"], {
...
 });
 
+//
+/**
+ * Tagged template TR_URL`path`. Resolves against W3C's /TR/
+ * */
+function W3URL(strings, ...values) {
+  const str = strings.reduce((str, part, i) => {
+    return str + part + (values[i] || "");
+  }, "");
```
Comment (saschanaz) [inline src/w3c/headers.js:118]: I wonder whether map+join is more readable here.

**Review by saschanaz — COMMENTED** [review]
BTW, is this only to skip `level` from the URL? I wonder whether we should add an option to skip it instead of overriding the whole path 🤔

**Review by marcoscaceres — COMMENTED** [review]
Thanks for the great feedback @saschanaz. I think I'd like to keep allowing `latestVersion` to be overridden instead of introducing a new pref. We were already allowing it for BG/CG, so this just makes it consistent (+ it's one less special case to think about).

File: src/w3c/headers.js [hunk 6]
```diff
@@ -108,6 +107,14 @@ const W3CDate = new Intl.DateTimeFormat(["en-AU"], {
   day: "2-digit",
 });
 
+/**
+ * Tagged template. Resolves against https://www.w3c.org.
```
Comment (sidvishnoi) [inline src/w3c/headers.js:111]: Why not use a regular function instead?

Comment (saschanaz) [reply src/w3c/headers.js:111]: There are heavy uses of template string so I think it's better to be a template string function.

Comment (saschanaz) [reply src/w3c/headers.js:111]: Hmm no, it can just be a wrapper without actual need for TSF...

Comment (marcoscaceres) [reply src/w3c/headers.js:111]: yeah, a function is probably fine. Doesn't look as cool tho 😎🤪.

**Comment by marcoscaceres** [comment]
Added docs https://github.com/w3c/respec/wiki/latestVersion

**Comment by marcoscaceres** [comment]
Sent https://github.com/w3c/respec-web-services/pull/210

**Review by sidvishnoi — APPROVED** [review]

**Review by saschanaz — APPROVED** [review]
👍

File: src/w3c/headers.js [hunk 7]
```diff
@@ -108,6 +107,15 @@ const W3CDate = new Intl.DateTimeFormat(["en-AU"], {
   day: "2-digit",
 });
 
+/**
+ * Resolves against https://www.w3.org.
+ * @param {string} href
+ * @returns
```
Comment (saschanaz) [inline src/w3c/headers.js:113]: Oh BTW, `{URL}` or just remove this?

Comment (marcoscaceres) [reply src/w3c/headers.js:113]: Will remove. Returns a string.

Comment (marcoscaceres) [inline src/w3c/headers.js:113]:
```suggestion
```

**Pull request merged**